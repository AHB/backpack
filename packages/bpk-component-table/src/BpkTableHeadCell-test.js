/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2021 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow strict */

import React from 'react';
import { render } from '@testing-library/react';

import BpkTableHeadCell from './BpkTableHeadCell';

describe('BpkTableHeadCell', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<BpkTableHeadCell>Heading</BpkTableHeadCell>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly with "alternate"', () => {
    const { asFragment } = render(
      <BpkTableHeadCell alternate>
        <th>Skyscanner</th>
      </BpkTableHeadCell>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly with custom class', () => {
    const { asFragment } = render(
      <BpkTableHeadCell className="my-custom-class">
        <th>Skyscanner</th>
      </BpkTableHeadCell>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly with arbitrary props', () => {
    const { asFragment } = render(
      <BpkTableHeadCell id="my-custom-id" data-foo="bar">
        <th>Skyscanner</th>
      </BpkTableHeadCell>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
