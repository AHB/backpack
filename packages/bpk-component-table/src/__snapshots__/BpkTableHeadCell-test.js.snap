// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkTableHeadCell should render correctly 1`] = `
<DocumentFragment>
  <th
    class="bpk-table__cell bpk-table__cell--head"
  >
    Heading
  </th>
</DocumentFragment>
`;

exports[`BpkTableHeadCell should render correctly with "alternate" 1`] = `
<DocumentFragment>
  <th
    class="bpk-table__cell bpk-table__cell--head-alternate"
  />
  <th>
    Skyscanner
  </th>
</DocumentFragment>
`;

exports[`BpkTableHeadCell should render correctly with arbitrary props 1`] = `
<DocumentFragment>
  <th
    class="bpk-table__cell bpk-table__cell--head"
    data-foo="bar"
    id="my-custom-id"
  />
  <th>
    Skyscanner
  </th>
</DocumentFragment>
`;

exports[`BpkTableHeadCell should render correctly with custom class 1`] = `
<DocumentFragment>
  <th
    class="bpk-table__cell bpk-table__cell--head my-custom-class"
  />
  <th>
    Skyscanner
  </th>
</DocumentFragment>
`;
